src/app/fetch-status.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable, EventEmitter } from '@angular/core';
+import { HttpClient, HttpResponseBase } from '@angular/common/http';
+
+import { LinkData } from './link.service';
+import { BehaviorSubject } from 'rxjs';
+
+/**
+ * Internal interface to pair a status code with a boolean indicating success
+ * or failure
+ */
+interface Status {
+  code?: number;
+  ok?: boolean;
+}
+
+/**
+ * This service is responsible for checking the status of each link
+ */
+@Injectable({
+  providedIn: 'root'
```

Comment:
A service became a singleton when we specify [providedIn](https://angular.io/guide/singleton-services). Usually, this is what we want for service like this. However, you have multiple variables that are not supposed to be shareable between multiple components (e.g. activeRequest, statusMap).

If you have multiple component with FetchStatus Service, the behavior became unpredictable. Is it possible to refactor this class to only contain shareable state/variable? (e.g. batchSize). One way to do it is to refactor initMap and startFetching into a single function.